### PR TITLE
Standardize on .NET Core v3.1

### PR DESCRIPTION
**Pull request by robmen**

### CODE REVIEW

**Comment by rseanhall** [comment]
```
C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(63,5): error NETSDK1083: The specified RuntimeIdentifier 'dne' is not recognized. [C:\projects\tools\src\wix\wix.csproj]
```

Just replace `dne` with a linux RID, like `rhel.7.1`.

**Comment by barnson** [comment]
Well that's batshirt crazy...

File: src/test/WixToolsetTest.Sdk/MsbuildUtilities.cs [hunk 1]
```diff
@@ -89,8 +89,7 @@ public static string GetQuotedPropertySwitch(BuildSystem buildSystem, string pro
         public static IEnumerable<string> GetToolCommandLines(MsbuildRunnerResult result, string toolName, string operation, BuildSystem buildSystem, bool? outOfProc = null)
         {
             var expectedOutOfProc = buildSystem == BuildSystem.DotNetCoreSdk || outOfProc.HasValue && outOfProc.Value;
-            var expectedToolExe = !expectedOutOfProc ? $"({toolName}.exe)" :
-                                  buildSystem == BuildSystem.DotNetCoreSdk ? $"{toolName}.dll\"" : $"{toolName}.exe";
+            var expectedToolExe = !expectedOutOfProc ? $"({toolName}.exe)" : $"{toolName}.exe";
```
Comment (robmen) [inline src/test/WixToolsetTest.Sdk/MsbuildUtilities.cs:92]: @rseanhall I did not fully dig into why this change was necessary to make the tests pass. Your review of it would be appreciated. 😄

Comment (rseanhall) [reply src/test/WixToolsetTest.Sdk/MsbuildUtilities.cs:92]: The .exe is architecture specific. The .net core SDK needs to do "dotnet.exe (app).dll" so it will work on all platforms.

Comment (rseanhall) [reply src/test/WixToolsetTest.Sdk/MsbuildUtilities.cs:92]: We want framework-dependent deployment: https://docs.microsoft.com/en-us/dotnet/core/deploying/deploy-with-cli. As you can see there, the behavior of our command to publish that way changed from 2.1 to 3.1. To keep the same behavior as we had with 2.1 where no .exe was built, we should either add this to the command line: `-p:UseAppHost=false` or add a property to the .csproj: `<UseAppHost>false</UseAppHost>`.

Comment (rseanhall) [reply src/test/WixToolsetTest.Sdk/MsbuildUtilities.cs:92]: If you want to do some more build work, we should be able to greatly simplify our build script now that they have implemented https://github.com/dotnet/sdk/issues/1675#issuecomment-784489066.

Comment (robmen) [reply src/test/WixToolsetTest.Sdk/MsbuildUtilities.cs]: Adding the `-p:UseAppHost=false` did resolve the problem. Thanks for pointing out the breaking change.

PS: I am very interested in the additional work to clean up the build script but I'll come back to that when doing build work for signing.

**Review by rseanhall — APPROVED** [review]
